frontend/src/hooks/api/useGetRecipeByUuid.api.ts

```diff
@@ -4,12 +4,15 @@ import api from "../../api";
 
 export const USE_GET_RECIPE_BY_UUID_KEY = "useGetRecipeByUuidKey";
 
+const queryFn = async ({ queryKey }: { queryKey: [string, string] }) => {
+  const [, uuid] = queryKey;
+  const response = await api.get<GetRecipeByIdRes>(`/recipe/${uuid}`);
+  return response.data.recipe;
+};
+
 export const useGetRecipeByUuid = (uuid: string) => {
   return useQuery({
     queryKey: [USE_GET_RECIPE_BY_UUID_KEY, uuid],
-    queryFn: () =>
-      api
-        .get<GetRecipeByIdRes>(`/recipe/${uuid}`)
-        .then((response) => response.data.recipe),
+    queryFn,
```

Comment:
```suggestion
    queryFn: ()=> queryFn(uuid),
```
ואז למעלה אתה מקבל את זה רגיל כמו כל פונקציה
ותחליף שמות מqueryFn לשם אינפורמטיבי פר שליפה
רלוונטי לכל הapis